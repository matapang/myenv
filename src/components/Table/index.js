import React from 'react';
import PropTypes from 'prop-types';
const Table = ({rows = [], columns = []}) => {    
    return (
        <table className="pt-html-table pt-interactive">
            <thead>
                <tr className="border-bottom">
                    {columns.map((c,key)=> <th key={key} >{c.headerName}</th>)}
                </tr>
            </thead>
            <tbody>
                {rows.map((r,rKey)=> <tr key={rKey}>
                     {columns.map((c,key)=> <td key={key} className={`border-top ${key === columns.length - 1 ? "w-100" :""}`} >{r[c.name]}</td>)}
                
                </tr>)}
            </tbody>
      </table>          
    )
};

Table.propTypes = {
    rows:PropTypes.arrayOf(PropTypes.object),
    columns:PropTypes.arrayOf(PropTypes.shape({name:PropTypes.string, headerName:PropTypes.string}))
}

export default Table;