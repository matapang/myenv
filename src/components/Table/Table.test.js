import React from 'react';

import renderer from 'react-test-renderer';
import Table from './index';


it('Table should match snapshot', () => {  
    const component = renderer.create(
      <Table rows={[{name:'Michael',  lastName:'Jordan'}]} columns={[{name:'name', headerName:'First Name'}, {name:'lastName', headerName:'Last Name'}]} />,
    );
    let tree = component.toJSON();
    expect(tree).toMatchSnapshot()
});
