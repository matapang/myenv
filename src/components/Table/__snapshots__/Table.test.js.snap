// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Table should match snapshot 1`] = `
<table
  className="table table-responsive .modifier"
>
  <thead>
    <tr>
      <th>
        First Name
      </th>
      <th>
        Last Name
      </th>
    </tr>
  </thead>
  <tbody>
    <tr>
      <td
        className="border-top"
      >
        Michael
      </td>
      <td
        className="border-top"
      >
        Jordan
      </td>
    </tr>
  </tbody>
</table>
`;
